base/random/RNGs.jl

```diff
@@ -0,0 +1,424 @@
+# This file is a part of Julia. License is MIT: https://julialang.org/license
+
+## RandomDevice
+
+const BoolBitIntegerType = Union{Type{Bool},Base.BitIntegerType}
+const BoolBitIntegerArray = Union{Array{Bool},Base.BitIntegerArray}
+
+if Sys.iswindows()
+    struct RandomDevice <: AbstractRNG
+        buffer::Vector{UInt128}
+
+        RandomDevice() = new(Vector{UInt128}(1))
+    end
+
+    function rand(rd::RandomDevice, T::BoolBitIntegerType)
+        rand!(rd, rd.buffer)
+        @inbounds return rd.buffer[1] % T
+    end
+
+    function rand!(rd::RandomDevice, A::BoolBitIntegerArray)
+        ccall((:SystemFunction036, :Advapi32), stdcall, UInt8, (Ptr{Void}, UInt32),
+              A, sizeof(A))
+        A
+    end
+else # !windows
+    struct RandomDevice <: AbstractRNG
+        file::IOStream
+        unlimited::Bool
+
+        RandomDevice(unlimited::Bool=true) =
+            new(open(unlimited ? "/dev/urandom" : "/dev/random"), unlimited)
+    end
+
+    rand(rd::RandomDevice, T::BoolBitIntegerType)   = read( rd.file, T)
+    rand!(rd::RandomDevice, A::BoolBitIntegerArray) = read!(rd.file, A)
+end # os-test
+
+"""
+    RandomDevice()
+
+Create a `RandomDevice` RNG object. Two such objects will always
+generate different streams of random numbers.
+"""
+RandomDevice
+
+### generation of floats
+
+rand(rng::RandomDevice, ::Type{Close1Open2}) =
+    reinterpret(Float64, 0x3ff0000000000000 | rand(rng, UInt64) & 0x000fffffffffffff)
+
+rand(rng::RandomDevice, ::Type{CloseOpen}) = rand(rng, Close1Open2) - 1.0
+
+@inline rand(r::RandomDevice, T::BitFloatType) = rand_generic(r, T)
```

Comment:
OK, I thought it should be harmless (non-breaking), but will amend. Other changes are: removing of the `win32_SystemFunction036!` (incorporated in `rand!(::MersenneTwister)`), and removing of the never-used never-exported `globalRNG()` function: do you want me to also split that out in a separate PR?